solr/core/src/java/org/apache/solr/cli/PostTool.java

```diff
@@ -330,15 +331,24 @@ public void runImpl(CommandLine cli) throws Exception {
       }
 
       solrUrl = SolrCLI.normalizeSolrUrl(solrUrl, true, hostContext) + hostContext;
-      String url = solrUrl + "/" + cli.getOptionValue("name") + "/update";
-      solrUpdateUrl = new URI(url);
+
+      solrUpdateUrl =
+          UriBuilder.fromUri(SolrCLI.normalizeSolrUrl(solrUrl, true, hostContext))
+              .path(hostContext)
+              .path(cli.getOptionValue("name"))
+              .path("update")
+              .build();
 
     } else if (cli.hasOption("solr-update-url")) {
       String url = cli.getOptionValue("solr-update-url");
       solrUpdateUrl = new URI(url);
     } else if (cli.hasOption("name")) {
-      String url = SolrCLI.getDefaultSolrUrl() + "/solr/" + cli.getOptionValue("name") + "/update";
-      solrUpdateUrl = new URI(url);
+      solrUpdateUrl =
+          UriBuilder.fromUri(SolrCLI.getDefaultSolrUrl())
+              .path("solr")
```

Comment:
Here, when `--solr-url` or `--solr-update-url` is not given, meaning it will be localhost, we always use `/solr`. But what if the user is at an installation with customized `contextPath` (I doubt that happens very often)?

Reply:
That is a good question. Eventually out of context of this PR, but a case we should eventually test. Seems like custom contexts in general may not have been tested and documented enough.